### PR TITLE
Get ECR auth details and refactor get account id

### PR DESCRIPTION
**Pull request by NishanthSpShetty**
# Description

Get elastic registry auth details to push docker images to  user registry account.  Fetch ecr url and token for the given account. Refactor get account id.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# why
- Get aws ecr url and token.

# How Has This Been Tested?

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: cmd/testclient/client.go [hunk 1]
```diff
@@ -489,6 +489,17 @@ func main() {
 			os.Exit(1)
 		}
 		sugar.Infow("GetWorkspaceCost method", "response", v)
+	case "GetElasticRegistryAuth":
```
Comment (sid255) [inline cmd/testclient/client.go:492]: Let's name this `GetContainerRegistryAuth`

File: pkg/service/aws/ecr.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/pkg/errors"
+	proto "gitlab.com/netbook-devs/spawner-service/proto/netbookai/spawner"
+)
+
+func (a *AWSController) GetElasticRegistryAuth(ctx context.Context, req *proto.GetElasticRegistryAuthRequest) (*proto.GetElasticRegistryAuthResponse, error) {
```
Comment (sid255) [inline pkg/service/aws/ecr.go:11]: Let's rename this too

File: pkg/service/aws/ecr.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	}
+
+	a.logger.Debug(ctx, "fetched accountId", "id", account_id)
+
+	//get ecr token
+	ecrClient := session.getEcrClient()
+
+	// Doc : https://docs.aws.amazon.com/sdk-for-go/api/service/ecr/#ECR.GetAuthorizationToken
+	res, err := ecrClient.GetAuthorizationTokenWithContext(ctx, &ecr.GetAuthorizationTokenInput{
+		RegistryIds: []*string{&account_id},
```
Comment (sid255) [inline pkg/service/aws/ecr.go:35]: Were you able to test this? Seems weird that we are sending AWS account id as Registry Id

**Review by sid255 — CHANGES_REQUESTED** [review]

File: pkg/service/aws/ecr.go (hunk 3 above)
Comment (NishanthSpShetty) [reply pkg/service/aws/ecr.go:35]: yes, account Id is registryId in ECR context

**Review by sid255 — APPROVED** [review]
LGTM

**Pull request merged**